resources/nspawn.go

```diff
@@ -0,0 +1,351 @@
+// state. This is run before Watch and again if watch finds a change occurring to the state.
+func (obj *NspawnRes) CheckApply(apply bool) (checkok bool, err error) {
+	log.Printf("%v[%v]: CheckApply(%t)", obj.Kind(), obj.GetName(), apply)
+
+	if obj.isStateOK { // cache the state
+		return true, nil
+	}
+
+	// This resource depends on systemd. Ensure that it's running.
+	// TODO: If systemd is not running, should we start it?
+	// TODO: We do this in two different places. Should we do it in Init?
+	if !systemdUtil.IsRunningSystemd() {
+		return false, errors.New("Systemd is not running.")
+	}
+
+	// Connect to org.freedesktop.machine1.Manager
+	conn, err := machined.New()
+
+	if err != nil {
+		return false, fmt.Errorf("Failed to connect to dbus: %s", err)
+	}
+
+	// Compare the current state with the desired state and perform the
+	// appropriate action
+	properties, err := conn.GetProperties(obj.GetName())
+	if err != nil {
+		if err.(dbus.Error).Name != "org.freedesktop.machine1.NoSuchMachine" {
+			log.Printf("GetProperties error: %s", err.Error())
+			return false, err
+		}
+		// If we could not successfully get the properties because there's no such machine, the machine
+		// is stopped.
+		if obj.State == stopped {
+			properties = make(map[string]interface{}, 1)
+			properties["State"] = stopped
+		}
+		// error if we need the image, ignore if we don't
+		_, err = conn.GetImage(obj.GetName())
+		if err != nil && obj.State != stopped {
+			return false, fmt.Errorf("No machine nor image named '%s'", obj.GetName())
+		}
+	}
+
+	// If the state matches, we're done.
+	if properties["State"] == obj.State {
+		return true, nil
+	}
+
+	if obj.State == running {
+		// Start the machine using svc resource
+		var serviceName = fmt.Sprintf(nspawnServiceTemplate, obj.GetName())
+		log.Printf("Start service: %s", serviceName)
+		service, err := NewSvcRes(serviceName, "running", "")
+		if err != nil {
+			return false, err
+		}
+		return service.CheckApply(true)
+	}
+	if obj.State == stopped {
+		// Terminate the machine with org.freedesktop.machine1.Manager.KillMachine
+		log.Printf("Stopping nspawn machine: %s", obj.GetName())
+		err = conn.KillMachine(obj.GetName())
+	}
+	return false, nil // If we didn't succeed, we failed.
+}
+
+// NspawnUID is a unique resource identifier
+type NspawnUID struct {
+	// NOTE: there is also a name variable in the BaseUID struct, this is
+	// information about where this UID came from, and is unrelated to the
+	// information about the resource we're matching. That data which is
+	// used in the IFF function, is what you see in the struct fields here.
+	BaseUID
+	name string // the machine name
+}
+
+// IFF aka if and only if they are equivalent, return true. If not, false.
+func (obj *NspawnUID) IFF(uid ResUID) bool {
+	res, ok := uid.(*NspawnUID)
+	if !ok {
+		return false
+	}
+	return obj.name == res.name
+}
+
+// NspawnResAutoEdges holds the state of the auto edge generator.
+type NspawnResAutoEdges struct {
```

Comment:
All this autoedges code isn't being used, so remove it. (Should probably look how it looks in noop.go)